### ISSUE TITLE
server/client freeze if global message on minetest.register_on_newplayer

### ISSUE THREAD

**Issue by verymilan** [minetest/minetest#3563]
Hi, i have some trouble using your api.
I am unable to send global messages using `minetest.register_on_newplayer`.
The player (singleplayer also) just joins into a gray world, without inventory.
The Server (or client if singleplayer) doesnt reacting anymore and have to be killed.
(cmd+c is working on server, but the client window stook).

It _does_ work, if i just send a private message, and it also does work, if i use 
`minetest.register_on_joinplayer` for example.

I tryed it for example like this:

```
local text = "test"
 minetest.register_on_newplayer(function()
 --minetest.after(1,function()
 minetest.chat_send_all(text)
 --end)
end)
```

^this example is working fine with `joinplayer`
Note: if i use core instead of minetest, it happends also.

---

Environments:
- server on the latest minetest master
- client on the latest arch linux release `0.4.13-4`

**Comment by rubenwardy** [comment]
It broke again, ffs. #2383
This is why we need Lua unit tests.

**Comment by rubenwardy** [comment]
is this still an issue?

**Comment by SmallJoker** [comment]
Yes, this is still a present bug. Using `minetest.chat_send_all` directly makes the server unresponsive, whereas it works fine using `minetest.after` to move the call into on_globalstep.

**Comment by nerzhul** [comment]
very strange, i think there is a deadlock somewhere, but the server lock is recursive

**Comment by nerzhul** [comment]
is this issue already valid ?

**Comment by SmallJoker** [comment]
Other issue: #8648
Fix: #8808
